app/src/main/java/me/saket/dank/ui/accountmanager/AccountManagerActivity.java

```diff
@@ -0,0 +1,290 @@
+        .refCount();
+
+    // Adapter data-set.
+    storedUsers
+        .toFlowable(BackpressureStrategy.LATEST)
+        .map(accounts -> {
+          List<AccountManagerScreenUiModel> uiModels = new ArrayList<>(accounts.size() + 1);
+
+          uiModels.add(AccountManagerPlaceholderUiModel.create());
+          uiModels.addAll(accounts);
+
+          return uiModels;
+        })
+        .compose(RxDiffUtil.calculateDiff(AccountManagerUiModelDiffer::create))
+        .observeOn(mainThread())
+        .doOnSubscribe(d -> loadingProgessbar.setVisibility(View.GONE))
+        .takeUntil(lifecycle().onDestroyFlowable())
+        .subscribe(accountManagerAdapter.get());
+
+    // Add new.
+    accountManagerAdapter.get().streamAddClicks()
+        .takeUntil(lifecycle().onDestroy())
+        .subscribe(o -> startActivity(LoginActivity.intent(this)));
+
+    // Drags.
+    ItemTouchHelper dragHelper = new ItemTouchHelper(createDragAndDropCallbacks());
+    dragHelper.attachToRecyclerView(accountsRecyclerView);
+    accountManagerAdapter.get().streamDragStarts()
+        .takeUntil(lifecycle().onDestroy())
+        .subscribe(viewHolder -> dragHelper.startDrag(viewHolder));
+
+    // Deletes.
+    // WARNING: THIS TOUCH LISTENER FOR SWIPE SHOULD BE REGISTERED AFTER DRAG-DROP LISTENER.
+    // Drag-n-drop's long-press listener does not get canceled if a row is being swiped.
+    accountsRecyclerView.addOnItemTouchListener(new RecyclerSwipeListener(accountsRecyclerView));
+    accountManagerAdapter.get().streamDeleteClicks()
+        .observeOn(io())
+        .flatMapCompletable(userToDelete -> queueToDelete(userToDelete))
+        .ambWith(lifecycle().onDestroyCompletable())
+        .subscribe();
+
+    // Switches.
+    accountManagerAdapter.get().streamSwitchClicks()
+        .observeOn(io())
+        .flatMapCompletable(userToSwitch -> this.userSessionRepository.get().switchAccount(userToSwitch.label()))
+        .ambWith(lifecycle().onDestroyCompletable())
+        .subscribe();
+
+    // Dismiss on outside click.
+    rootViewGroup.setOnClickListener(o -> finish());
+  }
+
+  private ItemTouchHelperDragAndDropCallback createDragAndDropCallbacks() {
+    return new ItemTouchHelperDragAndDropCallback() {
+      @Override
+      protected boolean onItemMove(@NonNull RecyclerView.ViewHolder source, @NonNull RecyclerView.ViewHolder target) {
+        AccountManagerViewHolder sourceViewHolder = (AccountManagerViewHolder) source;
+        AccountManagerViewHolder targetViewHolder = (AccountManagerViewHolder) target;
+
+        int fromPosition = sourceViewHolder.getAdapterPosition() - 1; // "Add Account placeholder will add 1 to Index
+        int toPosition = targetViewHolder.getAdapterPosition() - 1;
+
+        //noinspection ConstantConditions
+        List<AccountManager> accounts = Observable.fromIterable(accountManagerAdapter.get().getData())
+            .ofType(AccountManager.class)
+            .toList()
+            .blockingGet();
```

Comment:
Why are we creating an observable and then converting it back? Isn't it possible to avoid switching between these?

Reply:
Seems like it's for the `ofType` filtering functionality, I'll try to filter directly.

Reply:
I've filtered it with the `stream()` extension but I'm not sure it works down till API 23.